### PR TITLE
feat(detected_object_validation): add dynamic min pointcloud num

### PR DESCRIPTION
**Pull request by yukkysaito**
## Description

Because the validator used fixed values, the point cloud density was lower for distant obstacles, and obstacles were removed.
In this PR, the value is changed so that it changes according to the distance.
However, the parameters are not tuned and remain the same as before.

A similar approach is used in the following package.
https://github.com/autowarefoundation/autoware.universe/blob/main/perception/occupancy_grid_map_outlier_filter/include/occupancy_grid_map_outlier_filter/occupancy_grid_map_outlier_filter_nodelet.hpp#L63-L65

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: perception/detected_object_validation/src/obstacle_pointcloud_based_validator.cpp [hunk 1]
```diff
@@ -92,7 +92,10 @@ ObstaclePointCloudBasedValidator::ObstaclePointCloudBasedValidator(
   objects_pub_ = create_publisher<autoware_auto_perception_msgs::msg::DetectedObjects>(
     "~/output/objects", rclcpp::QoS{1});
 
-  min_pointcloud_num_ = declare_parameter<int>("min_pointcloud_num", 10);
+  points_num_threshold_param_.min_points_num = declare_parameter<int>("min_points_num", 10);
+  points_num_threshold_param_.max_points_num = declare_parameter<int>("max_points_num", 10);
+  points_num_threshold_param_.min_points_and_distance_ratio =
+    declare_parameter<float>("min_points_and_distance_ratio", 500.0);
```
Comment (yukkysaito) [inline perception/detected_object_validation/src/obstacle_pointcloud_based_validator.cpp:98]:
```suggestion
    declare_parameter<float>("min_points_and_distance_ratio", 800.0);
```

Comment (yukkysaito) [reply perception/detected_object_validation/src/obstacle_pointcloud_based_validator.cpp:98]: In the sample rosbag, it looked like it should be 10 points at 80m away.

**Comment by miursh** [comment]
@yukkysaito Could you add parameters to README?

**Comment by yukkysaito** [comment]
> @yukkysaito Could you write description about added parameters to README?

Thank you. I updated readme.
[b773498](https://github.com/autowarefoundation/autoware.universe/pull/2894/commits/b773498f859e65a8dabeaf5e28431c21982ce0e0)

**Review by yukke42 — APPROVED** [review]
LGTM

**Review by mitsudome-r — APPROVED** [review]

**Pull request merged**